src/verkle/algorithm.rs

```diff
@@ -0,0 +1,141 @@
+    headers
+}
+
+pub fn decode_block(rlp: String) -> Result<VerkleBlock, anyhow::Error> {
+    // rlp string starts with 0x
+    // catch panic?
+    let rlp_cropped: &str = &rlp[2..];
+    let serialized: Vec<u8> = hex::decode(rlp_cropped)?;
+    let block: VerkleBlock = decode(&serialized)?;
+
+    Ok(block)
+}
+
+// change to trace::
+#[allow(dead_code)]
+pub fn debug_block_info(block: &VerkleBlock) {
+    tracing::debug!(
+            "Block info:\n- parent hash: {}\n- storage root: {}\n- block number: {}\n",
+            hex::encode(block.header.parent_hash.clone()),
+            hex::encode(block.header.storage_root.clone()),
+            hex::encode(block.header.number.clone())
+        );
+    let keys = block.header.keyvals.keys.clone();
+    tracing::debug!("Kkey-vals: ");
+    for (indx, key) in keys.iter().enumerate() {
+        match block.header.keyvals.values[indx] {
+            Some(ref val) => tracing::debug!("\t{} => {}", hex::encode(key), hex::encode(val)),
+            None => tracing::debug!("\t{} is absent", hex::encode(key)),
+        }
+    }
+}
+
+pub async fn get_rlp(block_number: u64) -> Result<String, reqwest::Error> {
+    let client = Client::new();
+    let node_end_point: String = "https://rpc.condrieu.ethdevops.io/".to_owned();
+    let arg = format!(r#"{{"jsonrpc":"2.0","method":"debug_getBlockRlp","params":[{}],"id":"1"}}"#, block_number);
+
+    let res = client.post(&node_end_point)
+        .body(arg)
+        .headers(construct_headers())
+        .send()
+        .await?;
+
+    let block_rlp: RPCResp = res.json().await?;
+
+    Ok(block_rlp.result)
+}
+
+pub fn verification(block: VerkleBlock, parent_root: String) -> Result<verkle_trie::proof::UpdateHint, anyhow::Error> {
+    let parent_root = hex::decode(parent_root)?;
+    let root: EdwardsProjective = CanonicalDeserialize::deserialize(&parent_root[..])?;
+    let keyvals = block.header.keyvals;
+
+    let (checked, info) = block
+        .header
+        .proof
+        .verkle_proof
+        .check(keyvals.keys, keyvals.values, root);
+
+    match checked {
+        true => {
+            // change to trace::
+            tracing::info!("Good verification of block 0x{}", hex::encode(block.header.number));
+            match info {
+                Some(val) => Ok(val),
+                None => Err(anyhow::anyhow!("UpdateHint is none"))
+            }
+        },
+        false => {
+            // change to trace::
+            tracing::error!("Bad verification");
+            Err(anyhow::anyhow!("Verification didn't work out"))
+        }
+    }
+}
+
+// this function should use with tokio::task::spawn_blocking
+#[allow(dead_code)]
+pub fn save_rlp(rlp: String, num: u64) -> Result<(), anyhow::Error> {
+    use std::fs::File;
+    use std::io::prelude::*;
+
+    let mut file = File::create(format!("condrieu.block{}.rlp", num))?;
+    // catch panic?
+    let rlp_cropped = &rlp[2..];
+    file.write_all(&hex::decode(rlp_cropped)?)?;
+
+    Ok(())
+}
```

Comment:
Could you please remove functions that are not used anymore

Reply:
ok